interface/wx/webrequest.h

```diff
@@ -0,0 +1,292 @@
+// Name:        webrequest.h
+// Created:     2018-10-14
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxWebRequest
+
+    wxWebRequest allows for simple HTTP requests using the operating systems
+    components as implementation. The latest features of their operating
+    system will be used if available.
+
+    This class is well suited for simple HTTP requests.
+
+    This class supports synchronous and asynchronous usage. Synchronous usage
+    should only be used in secondary threads to not block the user interface.
+    When starting an asynchronous request via SendAsync() events need to be
+    handled.
+
+    Example for synchronous usage:
+
+    @code
+        wxWebRequest request("https://www.wxwidgets.org/downloads/logos/blocks.png");
+        if (request.Send())
+        {
+            // Load the response data into a wxImage
+            wxImage logoImage(request->GetResponseStream());
+            if (logoImage.IsOK())
+                wxLogInfo("Image loaded");
+        }
+        else
+        {
+            // Log the error
+            wxLogError("Could not load logo: %s", request.GetStatusText());
+        }
+    @endcode
+
+    System wide configuration like proxy and SSL certificates will be used
+    when possible.
+
+    @section descriptions Implementation Descriptions
+
+    The following APIs are used per platform, additional documentation
+    about supported features may be found in ther documentation
+
+    @par WinHTTP on Windows
+
+    This API is available on all supported Windows Versions and supports
+    HTTP and HTTPs up to HTTP/2.
+
+    <a href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">
+    API Documentation</a>
+
+    @par NSURLSession on macOS and iOS
+
+    This API is available since macOS 10.9 and iOS 7.0.
+
+    <a href="https://developer.apple.com/documentation/foundation/urlsession">
+    API Documentation</a>
+
+    @par libcurl on Linux
+
+    This library is available on most Unix distributions.
+
+    <a href="https://curl.haxx.se/libcurl/">
+    API Documentation</a>
+
+    @beginEventEmissionTable{wxWebRequestEvent}
+    @event{wxEVT_WEBREQUEST_READY(id, func)}
+        The request data is ready to be read
+    @event{wxEVT_WEBREQUEST_FAILED(id, func)}
+        The request has failed
+    @event{wxEVT_WEBREQUEST_AUTH_REQUIRED(id, func)}
+        The request needs additional authentication to continue
+    @endEventTable
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebRequestSession
+*/
+class wxWebRequest: public wxEvtHandler
+{
+public:
+    /**
+        Create a new request instance
+
+        @param url The URL of the HTTP resource for this request
+        @param session If no wxWebRequestSession is give the default session will
+                       be used
+
+        @since 3.1.2
```

Comment:
BTW, I don't really mind having them neither, but I don't think we need `@since 3.1.2` markers in all the methods of a class which is itself marked as being new in 3.1.2.